libimage/filters.go

```diff
@@ -20,31 +20,29 @@ import (
 // indicates that the image matches the criteria.
 type filterFunc func(*Image) (bool, error)
 
-// Apply the specified filters.  At least one filter of each key must apply.
+// Apply the specified filters.  All filters of each key must apply.
 func (i *Image) applyFilters(ctx context.Context, filters map[string][]filterFunc) (bool, error) {
```

Comment:
Other than handling of empty slices in the values of `filters`, AFAICS this is equivalent to
```
for …
  for … {
    m, err := filter() 
    if err != nil { … }
    if !m { return false }
  }
}
return true
```

with no need to maintain the `matches` and `groupMatches` variables.

And in the case of empty filter slices, I’m tempted to say that if they are allowed at all, they should result in `true` as the simpler code does (but I didn’t check what Docker does)

Reply:
This does the same thing, fixed! Thanks :)